### ISSUE TITLE
chmod not support change mode to  all files with *

### ISSUE THREAD

**Issue by eggcaker** [nushell/engine-q#826]
when ran `chmod 666 ./test/*.txt` to change all files in sub folder test, got error :
```
chmod: cannot access './test/*': No such file or directory
```
but `chmod 666 ./test/hello.txt` works.

**Comment by kubouch** [comment]
Does `chmod 666 ./test/*.txt` work in other shells?

I think what we're missing here is a glob expansion on externals.

**Comment by eggcaker** [comment]
it's working in bash ,just tested
```
(base) eggcaker@cloud:~$ ls -alF ./test/*.txt
-rw-rw-rw- 1 eggcaker eggcaker 0 Jan 23 06:59 ./test/hello.txt
(base) eggcaker@cloud:~$ chmod 444 ./test/*.txt
(base) eggcaker@Xiaoice:~$ ls -alF ./test/*.txt
-r--r--r-- 1 eggcaker eggcaker 0 Jan 23 06:59 ./test/hello.txt
```

**Comment by kubouch** [comment]
OK, then, I think we're lacking a universal glob expansion.

We would need to expand the glob ourselves and send the matching files to `chmod` (or any external) as a list (to be expanded as individual arguments, not just a long space-separated string). 

We would also need a way to turn it off.

**Comment by kubouch** [comment]
@eggcaker I think this workaround should work: `chmod 666 (ls ./test/*.txt).name`

**Comment by eggcaker** [comment]
@kubouch yes , it's works or longer version more nu's way ` ls ./test/*.txt |get name |each {chmod $it }, I'm just thinking maybe for those similar command it's working in another shells ,but also working the nushell with nushell style, just show some tips instead of a error, ex: 
```
your command is a bash format blabla , you should do with nushell  style:
chmod 666 (ls ./test/*.txt).name 
```

**Comment by jntrnr** [comment]
I think this might work in nushell and not e-q because e-q doesn't call through the shell anymore. Nushell would call through bash, which would do its own expansions of the glob patterns in the arguments.

**Comment by eggcaker** [comment]
actually not work in nushell either

**Comment by jntrnr** [comment]
This should now be fixed in the latest main with #839. Let me know if it works for you.

**Comment by eggcaker** [comment]
works !
